### ISSUE TITLE
Problem with BatchNormalization for CNNs

### ISSUE THREAD

**Issue by JCRMT-lima** [tensorflow/hub#537]
Hi, I'm trying to create a CNN - AlexNet with Keras, and I'm trying to use Batch Normalization for the first 2 layers (Using the Functional API).

I'm running a for loop to pick random variables from lists.
The problem is that when I try to run the loop with a simple line for batch normalization, I get an error: "BatchNormalization' object has no attribute 'shape'"

I don't know what the input shape is going to be since it is random each time.  

I somewhat new to coding, I'd be glad for any advice.

`for i in range(10):
  #Here are the four hyperparameters that we'll pick at random from the lists above
  a = random.choice(activation_list)
  h = random.choice(optimizer_list)
  batch = random.choice(batch_list)
  filters = random.choice(filters_list)
  full_neurons = random.choice(neurons_list)

  #Build model
  ##input
  visible = keras.Input(shape=(32,32,3))
  #1st layer
  conv1 = tf.keras.layers.Conv2D(filters,kernel_size = (3,3),strides = (1,1),activation = a, padding = 'same')(visible) #can randomize kernel_size, strides, padding
  pool1 = tf.keras.layers.MaxPooling2D(pool_size = (2,2), strides = (1,1))(conv1) #can randomize pool_size, strides
  batch_norm1 = tf.keras.layers.BatchNormalization()`

**Comment by rmothukuru** [comment]
@JCRMT-lima,
Looking at the code, the issue doesn't seem to be related to Tensorflow Hub. Can you please confirm if you are using TF Hub? If you are not using TF Hub, and if the problem is with Tensorflow, please raise the issue in this [Tensorflow Github Repository](https://github.com/tensorflow/tensorflow/issues/new/choose). Thanks!

**Comment by khimraj** [comment]
@JCRMT-lima, Can you share full code or link of notebook file to reproduce mentioned error.

**Comment by rmothukuru** [comment]
@JCRMT-lima,
Can you please provide the information asked above. Thanks!

**Comment by rmothukuru** [comment]
@JCRMT-lima,
Can you please provide the information asked above. Thanks!

**Comment by rmothukuru** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
